presto-main/src/main/java/com/facebook/presto/operator/InMemoryJoinHash.java

```diff
@@ -170,7 +183,7 @@ private boolean positionEqualsCurrentRow(int leftPosition, int rightPosition, Pa
         int blockIndex = decodeSliceIndex(pageAddress);
         int blockPosition = decodePosition(pageAddress);
 
-        return pagesHashStrategy.positionEqualsRow(blockIndex, blockPosition, rightPosition, rightPage);
+        return pagesHashStrategy.positionEqualsRowIgnoreNulls(blockIndex, blockPosition, rightPosition, rightPage);
```

Comment:
Yes, it should be named `positionEqualsCurrentRowIgnoreNulls`.

`WindowOperator` calls different method from `PagesHashStrategy`. There were previously two versions of `positionEqualsRow`. One taking blocks and marked as deprecated, second taking Page and `int[] rightChannels`. This second one is being used in window operator and it hasn't changed.

However you are right that I missed, that `PagesIndex` is being used in `WindowOperator`. I have partially fixed and partially refactored `WindowOperator`.